web/static/src/demo.js

```diff
@@ -0,0 +1,167 @@
+            } else {
+                return parsedText
+            }
+        }).join('');  // tokens already contain whitespace
+        // stop progressbar animation
+        clearInterval(timer);
+        smoothProgressbarComplete(progressBar);
+    }).catch(error => {
+        console.log(error);
+        clearInterval(timer);
+        progressBar.value = 0;
+    });
+}
+
+function parseTableRow(rowItemsAsList, itemElement='td', itemsClasses='', firstItemClasses='') {
+    const parsedItems = rowItemsAsList.map((item, idx) => {
+        const classList = idx === 0 ? [itemsClasses, firstItemClasses].join(' ') : itemsClasses;
+        return `<${itemElement} class="${classList}">${item}</${itemElement}>`;
+    }).join(' ');
+    return `<tr> ${parsedItems} </tr>`;
+}
+
+function similarityHandler(inputElement, outputElement) {
+    // set up progressbar animation
+    const progressBar = document.getElementById('vectors-similarity-progressbar')
+    progressBar.value = 0;
+    let timer = setInterval(() => {
+        smoothProgressbarIncrement(progressBar);
+    }, 50);
+    // query similarity endpoint
+    axios.post('/similarity-demo', {
+        'words': inputElement.value
+    }).then(response => {
+        console.log(response);
+        const headers = parseTableRow(Array.from(response.data['th']), 'td', 'uk-text-bold', '');
+        const tableData = Array.from(response.data['td'])
+            .map(rowOfItems => parseTableRow(Array.from(rowOfItems), 'td', '', 'uk-text-bold'))
+            .join('\n');
+        outputElement.innerHTML = `
+            ${headers}
+            ${tableData}
+        `;
+        // stop progressbar animation
+        clearInterval(timer);
+        smoothProgressbarComplete(progressBar);
+    }).catch(error => {
+        console.log(error);
+        clearInterval(timer);
+    });
+}
+
+function addDemoEventListener(triggers, handlerFunction, inputElementId, outputElementId) {
+    const inputEl = document.getElementById(inputElementId);
+    const outputEl = document.getElementById(outputElementId);
+    const boundHandler = handlerFunction.bind(null, inputEl, outputEl);
+    triggers.forEach(triggerData => {
+        document.getElementById(triggerData.elementId).addEventListener(triggerData.eventName, e => {
+            e.preventDefault();
+            boundHandler();
+            return false;
+        });
+    });
+    boundHandler();  // execute handler for the 1st time on registration
+}
+
+function addDemoEventListenerWithDefaulNaming(demoEventName, handler) {
+    return addDemoEventListener(
+        [{
+            elementId: `${demoEventName}-submit`,
+            eventName: 'click'
+        },
+        {
+            elementId: `${demoEventName}`,
+            eventName: 'submit'
+        }],
+        handler,
+        `${demoEventName}-input`,
+        `${demoEventName}-output`
+    );
+}
+
+// UIkit.util.ready(() => {
```

Comment:
Do we need this code?

Reply:
It will be useful when merging with the other demo.